### PR TITLE
[RNMobile][Embed block] Update bottom sheet to show specific embed label

### PR DESCRIPTION
**Pull request by ceyhun**
## Description

Updated bottom sheet UI to show specific embed label (like Youtube, Twitter, etc.)

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3961

## How has this been tested?

1. Use `/` inserter to add a specific embed block (like Youtube, Twitter, etc.)
2. Press ADD LINK button to open bottom sheet
3. Observe that the label for adding a link uses the specific embed block title (like Youtube, Twitter, etc.)
4. Try the same with a generic Embed block and the label should say `Embed link` 

## Screenshots 

Before | After
-- | -- 
|

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by fluiddot — COMMENTED** [review]
The changes look great to me, but I found a couple of non-critical issues that we might address before merging. Let me know if you'd like to address them, if not I'll be more than happy to approve the PR.

### WordPress embeds are not showing a specific label

As it happened when filtering the inline preview by providers ([reference](https://github.com/WordPress/gutenberg/blob/3ba07e547d306d327eb01a3f95ca2ba82d5cbe4f/packages/block-library/src/embed/edit.native.js#L216-L217)), we would need to handle this specific case and set the label `WordPress` for this embed type.



### Embed label reduces the URL field size

I think this is a known issue, but now that the label can be longer depending on the provider's name, the URL field can get shorter. I wouldn't block the PR due to this but it would be nice if we prioritize a potential fix and open an issue as a follow-up.

**Comment by ceyhun** [comment]
Thanks for the review @fluiddot 🙇 

> WordPress embeds are not showing a specific label


Seems like this is only reproducible when a WP link is set on the embed block, the preview is shown and when the link is edited. I works fine when we add a WP embed block the first them and its placeholder is shown. I've handled this case in https://github.com/WordPress/gutenberg/pull/34844/commits/55575b146284738d89e1287e81e5739890af9903.


Placeholder | Preview
-- | -- 
|


> Embed label reduces the URL field size

I think this might be fixed by the new bottom sheet implementation @twstokes is working on: https://github.com/wordpress-mobile/gutenberg-mobile/issues/3819. Not sure if that one will only cover Audio Block or if we'll expand that to other places where a URL is inserted like here in the Embed Block.

**Comment by fluiddot** [comment]
> Seems like this is only reproducible when a WP link is set on the embed block, the preview is shown and when the link is edited. I works fine when we add a WP embed block the first them and its placeholder is shown. I've handled this case in [55575b1](https://github.com/WordPress/gutenberg/commit/55575b146284738d89e1287e81e5739890af9903).

Right, I only managed to reproduce this way either, thanks for addressing it ❤️ !

> I thing this might be fixed by the new bottom sheet implementation @twstokes is working on: [wordpress-mobile/gutenberg-mobile#3819](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3819). Not sure if that one will only cover Audio Block or if we'll expand that to other places where a URL is inserted like here in the Embed Block.

It would be great if we expand it to other places like the embed block, @twstokes I'll be more than happy to integrate it in this block once it's ready, please let me know when the related PR is ready so I can test it, thanks 🙇 !

File: packages/block-library/src/embed/edit.native.js
Comment (fluiddot) [inline packages/block-library/src/embed/edit.native.js]: Nitpick: I see we're using the string `wp-embed` now in a couple of places so I'm wondering if we could have it as a constant instead, wdyt?

File: packages/block-library/src/embed/embed-bottom-sheet.native.js
Comment (fluiddot) [inline packages/block-library/src/embed/embed-bottom-sheet.native.js:40]: I'm wondering if we could memoize this with `useMemo` hook, I think it will help us preventing potential re-renderings as the component creates a new object on every pass.

Comment (fluiddot) [reply packages/block-library/src/embed/embed-bottom-sheet.native.js:40]: I tried adding `useMemo` here and looks like it doesn't reduce render calls, I don't have a strong feeling regarding whether to add it or not so I let you decide on this 😅 .

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I've just added a couple of nitpicking comments, feel free to omit them if you think it's ready to merge 👍

**Comment by twstokes** [comment]
> It would be great if we expand it to other places like the embed block, @twstokes I'll be more than happy to integrate it in this block once it's ready, please let me know when the related PR is ready so I can test it, thanks 🙇 !

Will do @fluiddot!

File: packages/block-library/src/embed/edit.native.js
Comment (ceyhun) [reply packages/block-library/src/embed/edit.native.js]: Makes sense, added it in d917d6ff8f3fb7c3ab6371b8cd031aaf111a1929 👍

File: packages/block-library/src/embed/embed-bottom-sheet.native.js
Comment (ceyhun) [reply packages/block-library/src/embed/embed-bottom-sheet.native.js:40]: I think this component won't be on screen most of the time and re-renders shouldn't be too expensive, so I'll skip it for now, until we perceive any performance issues with this 😅

**Pull request merged**
